lib/datasource/npm/get.ts

```diff
@@ -93,6 +94,27 @@ export async function getDependency(
     if (res.repository?.directory) {
       dep.sourceDirectory = res.repository.directory;
     }
+
+    // Massage the repository URL for non-complaint strings (see issue #4610)
+    // Remove the non-complaint segments of path, so the URL looks like "<scheme>://<domain>/<vendor>/<repo>"
+    // and add directory to the repository
+    const sourceUrlCopy = `${sourceUrl}`;
+    const sourceUrlSplit: string[] = sourceUrlCopy.split('/');
+
+    if (sourceUrlSplit.length > 5) {
```

Comment:
Is this only done for github.com or for all hosts?

Reply:
I think for all hosts